### PR TITLE
Sidebar ios bottom-bar compensation

### PR DESCRIPTION
**Pull request by sriramkrish85**

### CODE REVIEW

File: extensions/amp-sidebar/0.1/amp-sidebar.js [hunk 1]
```diff
@@ -26,6 +27,9 @@ const EXPERIMENT = 'amp-sidebar';
 /** @const */
 const TAG = 'AmpSidebar';
 
+/** @const */
+const IOS_SAFARI_BOTTOMBAR_HEIGHT = '10vh';
```
Comment (dvoytenko) [inline extensions/amp-sidebar/0.1/amp-sidebar.js]: This seems very suspicious. I pretty sure iOS defines these things in plane pixels. No?

Comment (dvoytenko) [inline extensions/amp-sidebar/0.1/amp-sidebar.js]: Also add a reference to the source doc.

File: extensions/amp-sidebar/0.1/amp-sidebar.js [hunk 2]
```diff
@@ -185,6 +203,59 @@ export class AmpSidebar extends AMP.BaseElement {
       });
     }
   }
+
+  /**
+   * @private
+   */
+  compensateIosBottombar_() {
+    if (!this.bottomBarCompensated_) {
+      //Compensate for IOS safari bottom navbar.
```
Comment (dvoytenko) [inline extensions/amp-sidebar/0.1/amp-sidebar.js]: nit: space

File: extensions/amp-sidebar/0.1/amp-sidebar.js [hunk 3]
```diff
@@ -185,6 +203,59 @@ export class AmpSidebar extends AMP.BaseElement {
...
+          e.preventDefault();
+        }
+      }
+    });
+  }
+
+  /**
+   * @private
+   */
+  compensateIosBottombar_() {
```
Comment (dvoytenko) [inline extensions/amp-sidebar/0.1/amp-sidebar.js]: Is it so bad you have to compensate it twice? :)

Comment (sriramkrish85) [inline extensions/amp-sidebar/0.1/amp-sidebar.js]: lol :) i hate merging now!

File: extensions/amp-sidebar/0.1/amp-sidebar.js (hunk 2 above)
Comment (sriramkrish85) [inline extensions/amp-sidebar/0.1/amp-sidebar.js]: Done

File: extensions/amp-sidebar/0.1/amp-sidebar.js (hunk 1 above)
Comment (sriramkrish85) [inline extensions/amp-sidebar/0.1/amp-sidebar.js]: This is to make sure we compensate enough for both 6 and 6plus. I custom measured this (not available anywhere in the open) I will make the design doc public after final edits if you think that would be good.

**Comment by sriramkrish85** [comment]
PTAL

File: extensions/amp-sidebar/0.1/amp-sidebar.js [hunk 4]
```diff
@@ -72,6 +75,12 @@ export class AmpSidebar extends AMP.BaseElement {
     /** @private @const {boolean} */
     this.isIosSafari_ = platform.isIos() && platform.isSafari();
 
+    /** @private @const {boolean} */
+    this.bottomBarCompensated_ = false;
```
Comment (erwinmombay) [inline extensions/amp-sidebar/0.1/amp-sidebar.js]: shouldnt be a const

**Comment by dvoytenko** [comment]
LGTM after @erwinmombay 's fixes.

File: extensions/amp-sidebar/0.1/amp-sidebar.js (hunk 4 above)
Comment (sriramkrish85) [inline extensions/amp-sidebar/0.1/amp-sidebar.js]: Thanks! done!

**Pull request merged**